bundle/bundle.go

```diff
@@ -16,19 +16,20 @@ import (
 
 // Bundle is a CNAB metadata document
 type Bundle struct {
-	SchemaVersion    string                 `json:"schemaVersion" mapstructure:"schemaVersion"`
-	Name             string                 `json:"name" mapstructure:"name"`
-	Version          string                 `json:"version" mapstructure:"version"`
-	Description      string                 `json:"description" mapstructure:"description"`
-	Keywords         []string               `json:"keywords,omitempty" mapstructure:"keywords"`
-	Maintainers      []Maintainer           `json:"maintainers,omitempty" mapstructure:"maintainers"`
-	InvocationImages []InvocationImage      `json:"invocationImages" mapstructure:"invocationImages"`
-	Images           map[string]Image       `json:"images,omitempty" mapstructure:"images"`
-	Actions          map[string]Action      `json:"actions,omitempty" mapstructure:"actions"`
-	Parameters       *ParametersDefinition  `json:"parameters,omitempty" mapstructure:"parameters"`
-	Credentials      map[string]Credential  `json:"credentials,omitempty" mapstructure:"credentials"`
-	Outputs          *OutputsDefinition     `json:"outputs,omitempty" mapstructure:"outputs"`
-	Definitions      definition.Definitions `json:"definitions,omitempty" mapstructure:"definitions"`
+	SchemaVersion      string                 `json:"schemaVersion" mapstructure:"schemaVersion"`
+	Name               string                 `json:"name" mapstructure:"name"`
+	Version            string                 `json:"version" mapstructure:"version"`
+	Description        string                 `json:"description" mapstructure:"description"`
+	Keywords           []string               `json:"keywords,omitempty" mapstructure:"keywords"`
+	Maintainers        []Maintainer           `json:"maintainers,omitempty" mapstructure:"maintainers"`
+	InvocationImages   []InvocationImage      `json:"invocationImages" mapstructure:"invocationImages"`
+	Images             map[string]Image       `json:"images,omitempty" mapstructure:"images"`
+	Actions            map[string]Action      `json:"actions,omitempty" mapstructure:"actions"`
+	Parameters         *ParametersDefinition  `json:"parameters,omitempty" mapstructure:"parameters"`
+	Credentials        map[string]Credential  `json:"credentials,omitempty" mapstructure:"credentials"`
+	Outputs            *OutputsDefinition     `json:"outputs,omitempty" mapstructure:"outputs"`
+	Definitions        definition.Definitions `json:"definitions,omitempty" mapstructure:"definitions"`
+	RequiredExtensions []string               `json:"requiredExtensions,omitempty" mapstructure:"requiredExtensions"`
```

Comment:
Since this slice can contain duplicate entries, should we add a test to cover that case and nail down what the behaviour should be?

Reply:
Ah, good point.  Should we just check that there are no duplicates during validation?  If there are duplicates, throw an error mentioning this?

Reply:
The slice can also contain items that are not present in the `custom` map, and we should probably validate that as well.

Reply:
> Ah, good point. Should we just check that there are no duplicates during validation? If there are duplicates, throw an error mentioning this?

Yes, that's probably better than issuing a warning and proceeding as that would encourage duplication/sloppiness.

Reply:
Updated; thanks!